src/Oscoin/Storage/Block/Abstract.hs

```diff
@@ -8,36 +8,60 @@
 
 module Oscoin.Storage.Block.Abstract
     ( BlockStore(..)
+    , FullBlockStoreAPI
+    , GetBlocksQuery(..)
 
     , hoistBlockStore
-    , noValidation
     , insertBlocksNaive
     , isNovelBlock
+
+    -- * Internals
+    , BlockStoreInternals(..)
+    , hoistBlockStoreInternals
     ) where
 
 import           Oscoin.Prelude
 
-import           Oscoin.Consensus.Types (Validate)
 import           Oscoin.Crypto.Blockchain (TxLookup)
 import           Oscoin.Crypto.Blockchain.Block
 import           Oscoin.Crypto.Hash (Hashed)
+import           Oscoin.Time.Chrono (NewestFirst, OldestFirst)
 
--- | An handle over a block storage backend.
-data BlockStore c tx s m = BlockStore
+-- The full 'BlockStore' API is the composition of its public and private parts.
+type FullBlockStoreAPI c tx s m =
+    ( BlockStore c tx s m
+    , BlockStoreInternals c tx s m
+    )
+
+data GetBlocksQuery c =
+      ByDepth Depth
+    -- ^ Get a collection of blocks from the input 'Depth' up to the tip.
+    | ByRootHash (BlockHash c)
+    -- ^ Get a collection of blocks starting from the input 'BlockHash' up to
+    -- the tip.
+
+-- | The internals for this 'BlockStore', i.e. the private API which shouldn't
+-- be visible to \"read-only consumers\".
+data BlockStoreInternals c tx s m = BlockStoreInternals
     { scoreBlock      :: Block c tx (Sealed c s) -> Score
```

Comment:
This should probably be moved to the protocol

Reply:
Yes, I should bite the bullet and be less lazy, I agree it doesn't belong here anymore.